src/main/java/com/structures/helpers/Structure.java

```diff
@@ -0,0 +1,422 @@
+        }
+    }
+
+    /**
+     * Renders the structure.
+     * @param startingPos the start pos to render.
+     * @param clientWorld the world of the client.
+     * @param player the player object.
+     * @param partialTicks the partial ticks.
+     */
+    public void renderStructure(@NotNull BlockPos startingPos, @NotNull final World clientWorld, @NotNull final EntityPlayer player, final float partialTicks)
+    {
+        Template.BlockInfo[] blockList = this.getBlockInfoWithSettings(this.settings);
+        Entity[] entityList = this.getEntityInfoWithSettings(clientWorld, startingPos, this.settings);
+
+        for (final Template.BlockInfo aBlockList : blockList)
+        {
+            Block block = aBlockList.blockState.getBlock();
+            IBlockState iblockstate = aBlockList.blockState;
+            BlockPos blockpos = aBlockList.pos.add(startingPos);
+            IBlockState iBlockExtendedState = block.getExtendedState(iblockstate, clientWorld, blockpos);
+            IBakedModel ibakedmodel = Minecraft.getMinecraft().getBlockRendererDispatcher().getModelForState(iblockstate);
+            TileEntity tileentity = null;
+            if (block.hasTileEntity(iblockstate))
+            {
+                tileentity = block.createTileEntity(clientWorld, iblockstate);
+                tileentity.readFromNBT(aBlockList.tileentityData);
+            }
+            ModelHolder models = new ModelHolder(blockpos, iblockstate, iBlockExtendedState, tileentity, ibakedmodel);
+            this.getQuads(models, models.quads);
+            this.renderGhost(clientWorld, models, player, partialTicks);
+        }
+
+        for (final Entity anEntityList : entityList)
+        {
+            Minecraft.getMinecraft().getRenderManager().renderEntityStatic(anEntityList, 0.0F, true);
+        }
+    }
+
+    private void renderGhost(final World world, ModelHolder holder, final EntityPlayer player, final float partialTicks)
+    {
+        boolean existingModel = !this.mc.theWorld.isAirBlock(holder.pos);
+
+        IBlockState actualState = holder.actualState;
+        Block block = actualState.getBlock();
+
+        if (actualState.getRenderType() == EnumBlockRenderType.MODEL)
+        {
+            BlockRenderLayer originalLayer = MinecraftForgeClient.getRenderLayer();
+
+            for (BlockRenderLayer layer : BlockRenderLayer.values())
+            {
+                if (block.canRenderInLayer(actualState, layer))
+                {
+                    this.mc.getTextureManager().bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+                    ForgeHooksClient.setRenderLayer(layer);
+                    this.renderGhostBlock(world, holder, player, layer, existingModel, partialTicks);
+                    holder.rendered = true;
+                }
+            }
+
+            ForgeHooksClient.setRenderLayer(originalLayer);
+        }
+
+        if (holder.te != null && !holder.rendered)
+        {
+            TileEntity te = holder.te;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'te' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)